clustermesh-apiserver/health/health.go

```diff
@@ -10,47 +10,45 @@ import (
 
 	"github.com/spf13/pflag"
 
-	"github.com/cilium/cilium/pkg/defaults"
+	"github.com/cilium/cilium/pkg/clustermesh/common"
 	"github.com/cilium/cilium/pkg/hive"
 	"github.com/cilium/cilium/pkg/hive/cell"
-	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
-	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "clustermesh-health")
+)
+
+const (
+	defaultHealthPort = 9880
+	optHealthPort     = "health-port"
 )
 
 type HealthAPIServerConfig struct {
-	ClusterMeshHealthPort int
+	HealthPort int
 }
 
-func (HealthAPIServerConfig) Flags(flags *pflag.FlagSet) {
-	flags.Int(option.ClusterMeshHealthPort, defaults.ClusterMeshHealthPort, "TCP port for ClusterMesh apiserver health API")
+func (def HealthAPIServerConfig) Flags(flags *pflag.FlagSet) {
+	flags.Int(optHealthPort, def.HealthPort, "TCP port for ClusterMesh apiserver health API")
 }
 
-var healthAPIServerCell = cell.Module(
+var DefaultHealthAPIServerConfig = HealthAPIServerConfig{
+	HealthPort: defaultHealthPort,
+}
+
+var HealthAPIServerCell = cell.Module(
 	"health-api-server",
 	"ClusterMesh Health API Server",
 
-	cell.Config(HealthAPIServerConfig{}),
-	cell.Provide(NewSyncState),
+	cell.Config(DefaultHealthAPIServerConfig),
 	cell.Invoke(registerHealthAPIServer),
 )
 
-func registerHealthAPIServer(lc hive.Lifecycle, clientset k8sClient.Clientset, cfg HealthAPIServerConfig, syncState *SyncState) {
+func registerHealthAPIServer(lc hive.Lifecycle, cfg HealthAPIServerConfig, syncState *common.SyncState) {
 	mux := http.NewServeMux()
 
-	mux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
As of now, the /healthz endpoint is unused. It's possible we may want to implement it, but they may also share the same logic (i.e. can't write to the local etcd). I think making it modular would be good future-proofing, though :+1:

Reply:
Yeah, I was mostly thinking about k8s client health checks which (as of now, although it may change in the future) would not be meaningful in the kvstoremesh context. But I agree that it is something not applicable right now, and we can always postpone generalization to later.